pygeohash/nbgeohash.py

```diff
@@ -0,0 +1,186 @@
+"""
+.. module:: nbgeohash
+   :platform: Unix, Windows
+   :synopsis: A module for encoding to and decoding from the geohash system
+
+.. moduleauthor:: ilyas Moutawwakil <ilyas.moutawwakil@gmail.com>
+
+
+"""
+
+
+from math import log10
+
+__base32 = "0123456789bcdefghjkmnpqrstuvwxyz"
+#  Note: the alphabet in geohash differs from the common base32 alphabet described in IETF's RFC 4648
+# (http://tools.ietf.org/html/rfc4648)
+
+try:
+    # Soft dependency
+    import numpy as np
+    from numba import njit, types
+except ImportError:
+    print("Numpy and Numba are soft dependencies to use this feature.")
+    raise ImportError(
```

Comment:
looks like this will still raise an import error, if it's a soft dependency it should just warn (warnings.warn maybe).

Reply:
I came from sktime where soft dependecies do raise an error. If it just warned the use of the njit decorator in code will raise the error anyways.

Reply:
been a while, but if you're still interested in merging this, i think this is the only open issue

Reply:
okay, I made it so that soft dependencies are checked on the first import of the module. If they are importable, functions from the nbgeohash module are added to the list, otherwise you just get a warning to only use native ones.

Reply:
I can also add them to the `extra_require` argument in setup:
```python
extras_require={
    'numba': ['numba', 'numpy'],
}
```
To be able to use `pip install pygeohash[numba]`.